include/base/libmesh_logging.h

```diff
@@ -45,6 +50,44 @@ class PerfLog;
 extern PerfLog perflog;
 
 
+/**
+ * Used for logging something that naturally lasts as long as some
+ * enclosing scope, such as the current function.  Makes it very easy
+ * to handle multiple return scenarios, since the event is popped in
+ * the destructor.  Should not be used directly, instead use the
+ * LOG_SCOPE macro, which resolves to nothing at compile time if
+ * logging is disabled.
+ *
+ * \author John Peterson
+ * \date 2016
+ */
+struct PerfItem
+{
+  PerfItem(const char * label,
+           const char * header,
+           bool enabled=true) :
+    _label(label),
+    _header(header),
+    _enabled(enabled)
+  {
+    if (_enabled)
+      libMesh::perflog.push(label, header);
```

Comment:
Should we hardcode libMesh::perflog here?  That's obviously what we want for the LOG_SCOPE macro, but it might be nice to let users instantiate this class with their own PerfLog object instead.